### PR TITLE
Video: ov7725: update to use gpio_dt_spec/i2c_dt_spec

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Review by loicpoulain — APPROVED** [review]

File: drivers/video/ov7725.c [hunk 1]
```diff
@@ -177,12 +177,12 @@ LOG_MODULE_REGISTER(ov7725);
 
 struct ov7725_config {
 	struct i2c_dt_spec i2c;
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	struct gpio_dt_spec reset_gpio;
+#endif
```
Comment (gmarull) [inline drivers/video/ov7725.c:182]: note: this was not coditional before

File: drivers/video/ov7725.c [hunk 2]
```diff
@@ -560,32 +560,23 @@ static const struct video_driver_api ov7725_driver_api = {
...
-
-		gpio_pin_set(drv_data->reset_gpio,
-			     drv_data->reset_pin,
-			     0);
-		k_sleep(K_MSEC(1));
-		gpio_pin_set(drv_data->reset_gpio,
-			     drv_data->reset_pin,
-			     1);
-		k_sleep(K_MSEC(1));
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
```
Comment (gmarull) [inline drivers/video/ov7725.c:568]: not multi-instance friendly

File: drivers/video/ov7725.c (hunk 1 above)
Comment (galak) [reply drivers/video/ov7725.c:182]: True, but the property is optional, so see no reason not to make it conditional to match video/ov2640.c

File: drivers/video/ov7725.c (hunk 2 above)
Comment (galak) [reply drivers/video/ov7725.c:568]: As @bbilas said, the driver isn't multi-instance friendly to begin with.  I'm also guessing its unlikely one would have more than one of these devices in a system.  (And if/when someone has such a system they will hopefully update the driver to support that).

File: drivers/video/ov7725.c (hunk 1 above)
Comment (bbilas) [reply drivers/video/ov7725.c:182]: It might be done in a separate commit but I have no objections here.

**Review by bbilas — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**